pages/api/web-scraping/flights.js

```diff
@@ -0,0 +1,187 @@
+// @ts-check
+import { FingerprintJsServerApiClient, Region } from '@fingerprintjs/fingerprintjs-pro-server-api';
+import { CheckResult, checkResultType } from '../../../server/checkResult';
+import { isRequestIdFormatValid, originIsAllowed, visitIpMatchesRequestIp } from '../../../server/checks';
+import { ALLOWED_REQUEST_TIMESTAMP_DIFF_MS, SERVER_API_KEY } from '../../../server/const';
+import { sendErrorResponse, sendForbiddenResponse, sendOkResponse } from '../../../server/response';
+import { ensurePostRequest, messageSeverity } from '../../../server/server';
+import { DAY_MS, FIVE_MINUTES_MS, HOUR_MS } from '../../../shared/const';
+import { AIRPORTS } from '../../web-scraping';
+
+/**
+ * @typedef {Object} ResultsQuery
+ * @property {string} from
+ * @property {string} to
+ * @property {string} requestId
+ * */
+
+/**
+ * @param {import('next').NextApiRequest} req
+ * @param {import('next').NextApiResponse} res
+ */
+export default async function getFlights(req, res) {
+  // This API route accepts only POST requests.
+  if (!ensurePostRequest(req, res)) {
+    return;
+  }
+
+  const { from, to, requestId } = /** @type {ResultsQuery} */ (req.body);
+
+  // Validate request ID format
+  if (!isRequestIdFormatValid(requestId)) {
+    sendForbiddenResponse(
+      res,
+      new CheckResult('Invalid request ID.', messageSeverity.Error, checkResultType.RequestIdMismatch)
+    );
+    return;
+  }
+
+  try {
+    // Retrieve analysis event from the Server API using the request ID
+    const client = new FingerprintJsServerApiClient({ region: Region.Global, apiKey: SERVER_API_KEY });
+    // If the requestId does not exist, the SDK will throw an error which will be caught below
+    const eventResponse = await client.getEvent(requestId);
+    const botData = eventResponse.products?.botd?.data;
+
+    if (!botData) {
+      sendOkResponse(
+        res,
+        new CheckResult(
+          'Bot detection is disabled, access allowed.',
+          messageSeverity.Success,
+          checkResultType.Passed,
+          getFlightResults(from, to)
+        )
+      );
+      return;
+    }
+
+    if (botData.bot?.result === 'good') {
+      sendOkResponse(
+        res,
+        new CheckResult(
+          'Access allowed, good bot detected.',
+          messageSeverity.Success,
+          checkResultType.GoodBotDetected,
+          getFlightResults(from, to)
+        )
+      );
+      return;
+    }
+
+    // Check for bot presence and type
+    if (botData.bot?.result === 'bad') {
```

Comment:
We check the `bad` and `good` values, bud trust that the rest variant would be `notDetected`. Sounds not 100% safe.

Reply:
Hmm my understanding of the API was that those are the only 3 possible values, so by definition it must be `notDetected`. How would you check for other unexpected values? 

Something like this?

```js
if (botData.bot?.result !== 'notDetected') {
sendForbiddenResponse(
      res,
      new CheckResult(
        'Server error, unexpected bot detection value',
        messageSeverity.Error,
        checkResultType.Error
      )
    );
    return;
} 
```

Reply:
Yes, look good. Because in future we can add another type of bot (I have no idea, but who knows)

Reply:
ok, done :)